identity/src/main/java/com/stripe/android/identity/ui/UploadScreen.kt

```diff
@@ -0,0 +1,366 @@
+                        ) {
+                            LoadingButtonState.Disabled
+                        } else {
+                            LoadingButtonState.Idle
+                        }
+                    )
+                }
+                LoadingButton(
+                    modifier = Modifier.testTag(UPLOAD_SCREEN_CONTINUE_BUTTON_TAG),
+                    text = stringResource(id = R.string.kontinue).uppercase(),
+                    state = continueButtonState
+                ) {
+                    continueButtonState = LoadingButtonState.Loading
+                    onContinueClicked()
+                }
+            }
+        }
+    }
+}
+
+internal const val FRONT_ROW_TAG = "frontRow"
+internal const val BACK_ROW_TAG = "backRow"
+internal const val UPLOAD_SCREEN_CONTINUE_BUTTON_TAG = "uploadScreenContinueButton"
+internal const val SHOULD_SHOW_TAKE_PHOTO_TAG = "shouldShowTakePhoto"
+internal const val SHOULD_SHOW_CHOOSE_PHOTO_TAG = "shouldShowChoosePhoto"
+
+internal enum class UploadMethod {
+    TAKE_PHOTO, CHOOSE_PHOTO
+}
+
+@Composable
+internal fun UploadImageDialog(
+    uploadInfo: DocumentUploadSideInfo,
+    onDismissRequest: () -> Unit,
+    onUploadMethodSelected: () -> Unit
+) {
+    Dialog(onDismissRequest = onDismissRequest) {
+        Surface(
+            shape = MaterialTheme.shapes.medium,
+            color = MaterialTheme.colors.surface,
+            contentColor = contentColorFor(backgroundColor = backgroundColor)
+        ) {
+            Column(
+                modifier = Modifier
+                    .padding(
+                        top = 16.dp,
+                        bottom = 16.dp,
+                        start = 24.dp,
+                        end = 24.dp
+                    )
+            ) {
+                Text(
+                    modifier = Modifier.padding(bottom = dimensionResource(id = R.dimen.item_vertical_margin)),
+                    text = getTitleFromScanType(scanType = uploadInfo.scanType),
+                    style = MaterialTheme.typography.subtitle1,
+                    fontWeight = FontWeight.Bold
+                )
+                if (uploadInfo.shouldShowTakePhoto) {
+                    TextButton(
+                        modifier = Modifier.testTag(SHOULD_SHOW_TAKE_PHOTO_TAG),
+                        contentPadding = PaddingValues(horizontal = 0.dp),
+                        onClick = {
+                            onUploadMethodSelected()
+                            uploadInfo.onPhotoSelected(UploadMethod.TAKE_PHOTO)
+                        }
+                    ) {
+                        Text(
+                            modifier = Modifier.fillMaxWidth(),
+                            text = stringResource(id = R.string.take_photo),
+                            color = MaterialTheme.colors.onBackground.copy(alpha = 0.6f),
+                            textAlign = TextAlign.Start
+                        )
+                    }
```

Comment:
To be more inline with the platform conventions, the options should be simple clickable list items instead of text buttons.

Since you’re building these types of dialogs in a few places, you could introduce a `DialogListItem` composable for a consistent look. What do you think?

```kotlin
@Composable
fun DialogListItem(
    text: String,
    onSelected: () -> Unit,
) {
    val listItemHeight = dimensionResource(
        com.google.android.material.R.dimen.abc_list_item_height_small_material
    )

    Box(
        modifier = Modifier
            .testTag(SHOULD_SHOW_TAKE_PHOTO_TAG)
            .height(listItemHeight)
            .clickable { onSelected() },
        contentAlignment = Alignment.CenterStart,
    ) {
        Text(
            modifier = Modifier.fillMaxWidth(),
            text = text,
            color = MaterialTheme.colors.onBackground.copy(alpha = 0.6f),
            textAlign = TextAlign.Start
        )
    }
}
```

Reply:
hmm the ripple effect of a `Box` won't follow the app's theme color(please see attached) - will keep using the TextButton to get all material theming but did refactor them into a `DialogListItem`

| after| before |
| ------------- | ------------- |
| ![c6b2c78d-675b-4272-a64d-3e83cf3446ee](https://user-images.githubusercontent.com/79880926/200956298-4959f605-9f31-4974-86bd-a8d12efb00c8.png) | ![ac133852-6142-4b3b-bc64-4ba24f30d228](https://user-images.githubusercontent.com/79880926/200956304-8c9ce8a7-2900-4349-9dcc-f889a6997698.png) |

Reply:
Items in alert dialogs typically don’t use the theme color, so that is fine. We’d just have to make sure that the ripple uses the entire width of the dialog. Can you tweak that?